phpBB/phpbb/search/backend/base.php

```diff
@@ -0,0 +1,481 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\search\backend;
+
+use phpbb\cache\service;
+use phpbb\config\config;
+use phpbb\db\driver\driver_interface;
+use phpbb\user;
+
+/**
+* optional base class for search plugins providing simple caching based on ACM
+* and functions to retrieve ignore_words and synonyms
+*/
+abstract class base implements search_backend_interface
+{
+	public const SEARCH_RESULT_NOT_IN_CACHE = 0;
+	public const SEARCH_RESULT_IN_CACHE = 1;
+	public const SEARCH_RESULT_INCOMPLETE = 2;
+
+	// Batch size for create_index and delete_index
+	private const BATCH_SIZE = 100;
+
+	/**
+	 * @var service
+	 */
+	protected $cache;
+
+	/**
+	 * @var config
+	 */
+	protected $config;
+
+	/**
+	 * @var driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var user
+	 */
+	protected $user;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param service $cache
+	 * @param config $config
+	 * @param driver_interface $db
+	 * @param user $user
+	 */
+	public function __construct(service $cache, config $config, driver_interface $db, user $user)
+	{
+		$this->cache = $cache;
+		$this->config = $config;
+		$this->db = $db;
+		$this->user = $user;
+	}
+
+	/**
+	 * Retrieves cached search results
+	 *
+	 * @param string $search_key an md5 string generated from all the passed search options to identify the results
```

Comment:
It looks like you tried to align these comments. If that's the case, please use tabs for alignment or don't align these at all. ;)